aip/general/0127.md

```diff
@@ -73,18 +73,21 @@ message CreateBookRequest {
     should be populated in the request proto. When extracting a [resource
     name][aip-122], the variable **must** include the entire resource name, not
     just the ID component.
+  - URIs **may** use nested fields for their variable names. (Additionally,
+    AIP-134 mandates this for `Update` requests.)
   - URIs **must** use the `*` character to represent ID components, which
     matches all URI-safe characters except for `/`. URIs **may** use `**` as
     the final segment of a URI if matching `/` is required.
-- The `body` key defines which field in the request will be sent as the HTTP
-  body. If the body is `*`, then this indicates that the request object itself
-  is the HTTP body. The request body is encoded as JSON as defined by protocol
-  buffers' canonical [JSON encoding][].
+- The `body` key defines which one field in the request will be sent as the
+  HTTP body. If the body is `*`, then this indicates that the request object
+  itself is the HTTP body. The request body is encoded as JSON as defined by
+  protocol buffers' canonical [JSON encoding][].
   - RPCs **must not** define a `body` at all for RPCs that use the `GET` or
     `DELETE` HTTP verbs.
   - RPCs **must** use the prescribed `body` for Create ([AIP-133][]) and Update
     ([AIP-134][]) requests.
   - RPCs **should** use the prescribed `body` for custom methods ([AIP-136][]).
+  - The `body` **must not** contain a nested field (or use the `.` character).
```

Comment:
I suggest we add notes about query params for repeated fields and for message fields (even if is worded as "MAY but SHOULD NOT")

Reply:
Done, but worded differently.

Reply:
I was actually referring to query params, saying `...?foo.a=A&foo.b=B` is allowed for sub-messages and `...?name=a&name=b` is allowed but discouraged for repeated fields.

Reply:
lmk what you think about mentioning the query params explicitly. Right now the only mention is in the request message comment.